### ISSUE TITLE
Actor multi animations always start at position (0,0,0)

### ISSUE THREAD

**Issue by bearney74** [panda3d/panda3d#1382]
## Description
I have 5 different animations for the same actor.  Each of these seem to start at (0,0,0).  After one animation finishes playing, I'd like for the next animation to start at the position where the last one stopped.  Currently, when I start a new animation, they all start at (0,0,0)
and just ignore where the last one stopped.
I have looked at the getPos function, but it always returns (0,0,0) for each "step" during the animation.  Any ideas?  

```
 self.actor = Actor("assets/Thriller0/Thriller0.gltf",
                           {
                           "0": "assets/Thriller0/Thriller0.gltf",
                           "1": "assets/Thriller0/Thriller0.gltf",
                           "2": "assets/Thriller1/Thriller1.gltf",
                           "3": "assets/Thriller2/Thriller2.gltf",
                           "4": "assets/Thriller3/Thriller3.gltf",
                           "5": "assets/Thriller4/Thriller4.gltf"
                           })
```

## Environment
  Linux (Lubuntu 22.10)   (and Windows 10)
  AMD 64
  Panda3d 1.10.12
  installed via pip
  Python 3.10.6

**Comment by rdb** [comment]
Animations do not affect the position of an object, only the positions of the armature's joints.

If you want to obtain the position of a particular joint, you should use exposeJoint to expose a joint as a node.

See:
https://docs.panda3d.org/1.10/python/programming/models-and-actors/attaching-objects-to-joints

**Comment by ArsThaumaturgis** [comment]
As rdb indicates, this behaviour is, I think, standard for animations.

As to a suggestion, if the animations always play out to their full, then, since animations are (presumably) unchanging, you could potentially keep a dictionary associating animation-names with final offsets. These could then be applied at the end of each animation via a call to "setPos". (Perhaps via a Sequence that first runs the animation, then calls a function that updates the position.)

**Comment by bearney74** [comment]
Okay.. Thanks guys.  I will try these ideas out and see if they are satisfactory.

**Comment by bearney74** [comment]
I"ve used exposeJoint to attach a node that I can track via getPos
But now when I try to set the actors Position, using setPos,  getPos returns the right value, but the animation still goes to (0,0,0).  Its like it is ignoring the position and using a preset value.  Do I need to go to each joint (segment) and modify its position?

Billy

**Comment by bearney74** [comment]
FYI..  The gltf files I am using are from mixamo.com.  I downloaded the models in fbx format and then used FBX2glTF-windows-x64.exe to convert them to gltf files.  I am not sure if that information will help, but thought I would throw that out there.

**Comment by bearney74** [comment]
Okay.. I think I figured this out.

I had to get the position using 'getPos' from the exposed joint, just like you guys suggested.   I put the initial position from this in a variable called position_initial.  Then when I set the position of the actor, I do actor.setPosition(render, position - position_initial), and then seems to set the actor in the some position as the last animation.    We need position_initial because it is not set to (0,0,0), like the actor is, and when I just set the position, the actor was moving each time a little further up, and left or right.   I can post some code if that will help someone else.

**Comment by bearney74** [comment]
Yes, using actor.setPos(...) seems to change the actor/animation position on the screen so that it appears where the previous animation ended.

Each time the update function is called it stores the current position in a variable called position.  When the animation stops,
I move onto the next animation and update the actors position so that it is set to the last known position.  This allows me to play the animations in any order.

Here's my update function that performs these tasks:
def update(self, task):
        _track = self.playlist[self.num]
    
        if self.actor.getAnimControl(_track).is_playing():
            self.position=self.actor_pos.getPos(self.render)
        else:
            self.num+=1
            _track = self.playlist[self.num]
            self.actor.setPos(self.render, self.position - self.position_initial)
            self.actor.play(_track)

        return task.cont


self.actor_pos is defined  in the __init__ function as 
self.actor_pos = self.actor.exposeJoint(None, "modelRoot", "mixamorig:Belly")

I went with the belly joint because I figured it maybe it would be the best joint to get a good position.  Trying another joint may improve results even more.

The issue that got me was that I was able to move the actor to whatever position I wanted but the getPos function never changed from (0,0,0).  Using the exposeJoint function helped solve that issue.

Thanks!
